clients/src/test/java/org/apache/kafka/clients/consumer/KafkaConsumerTest.java

```diff
@@ -2974,7 +2974,7 @@ public void testSubscriptionOnInvalidTopic(GroupProtocol groupProtocol) {
         KafkaConsumer<String, String> consumer = newConsumer(groupProtocol, time, client, subscription, metadata, assignor, true, groupInstanceId);
         consumer.subscribe(singleton(invalidTopicName), getConsumerRebalanceListener(consumer));
 
-        assertThrows(InvalidTopicException.class, () -> consumer.poll(Duration.ZERO));
+        assertThrows(InvalidTopicException.class, () -> consumer.poll(Duration.ofMillis(100)));
```

Comment:
I'm guessing you had to change the poll timeout here to make the test pass? Definitely a sign that we have some issues still. We should be able to keep the test as it was, passing with poll(0), as it does for the legacy consumer. I would suggest to address my concern above, and then review this to know if there is something else making the test fail with poll(0)